geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/ExportCommand.scala

```diff
@@ -187,12 +185,27 @@ trait ExportCommand[DS <: DataStore] extends DataStoreCommand[DS]
     }
   }
 
-  protected def export(ds: DS, query: Query, exporter: FeatureExporter): Option[Long] = {
+  /**
+   * Hook for overriding export
+   *
+   * @param ds data store
+   * @param query query
+   * @param exporter exporter
+   * @param empty export empty files or no
```

Comment:
Hmm... I wonder if there's a better variable name like `writeEmptyOutput` or `allowEmptyOutput`.  Admittedly, that's me being verbose.

It took me a few minutes of reading this PR to figure out what the `empty` flag was doing...  Thoughts?

Reply:
I'll update it